library/L1_Peripheral/stm32f10x/pwm.hpp

```diff
@@ -0,0 +1,296 @@
+#pragma once
+
+#include "L0_Platform/stm32f10x/stm32f10x.h"
+#include "L1_Peripheral/pwm.hpp"
+#include "L1_Peripheral/stm32f10x/pin.hpp"
+#include "L1_Peripheral/stm32f10x/system_controller.hpp"
+#include "utility/log.hpp"
+
+namespace sjsu::stm32f10x
+{
+/// An implementation of the sjsu::Pwm peripheral for the stm32f10x family of
+/// microcontrollers.
+class Pwm final : public sjsu::Pwm
+{
+ public:
+  static constexpr auto kOutputChannelEnable = bit::MaskFromRange(0);
+  static constexpr auto kTimerEnable         = bit::MaskFromRange(0);
+  static constexpr auto kUpdateGeneration    = bit::MaskFromRange(0);
+
+  // We use pins spread out across Timers 1 - 5
+  struct PwmPin_t
+  {
+    // Reference to the pin
+    sjsu::Pin & pin;
+
+    // Holds the address to the corresponding Timer peripheral
+    TIM_TypeDef * registers;
+
+    // Holds the channel (1 - 4) which corresponds to the pin
+    uint8_t channel : 3;
+
+    // Peripheral ID for the timer
+    sjsu::SystemController::ResourceID id;
+  };
+
+  struct PwmPin
+  {
+   private:
+    // Note: Pins on the same timer must use the same frequency
+    // Timer 1
+    inline static sjsu::stm32f10x::Pin pa8  = sjsu::stm32f10x::Pin('A', 8);
+    inline static sjsu::stm32f10x::Pin pa9  = sjsu::stm32f10x::Pin('A', 9);
+    inline static sjsu::stm32f10x::Pin pa10 = sjsu::stm32f10x::Pin('A', 10);
+    inline static sjsu::stm32f10x::Pin pa11 = sjsu::stm32f10x::Pin('A', 11);
+
+    // Timer 2
+    inline static sjsu::stm32f10x::Pin pa0 = sjsu::stm32f10x::Pin('A', 0);
+    inline static sjsu::stm32f10x::Pin pa1 = sjsu::stm32f10x::Pin('A', 1);
+    inline static sjsu::stm32f10x::Pin pa2 = sjsu::stm32f10x::Pin('A', 2);
+    inline static sjsu::stm32f10x::Pin pa3 = sjsu::stm32f10x::Pin('A', 3);
+
+    // Timer 3
+    inline static sjsu::stm32f10x::Pin pa6 = sjsu::stm32f10x::Pin('A', 6);
+    inline static sjsu::stm32f10x::Pin pa7 = sjsu::stm32f10x::Pin('A', 7);
+    inline static sjsu::stm32f10x::Pin pb0 = sjsu::stm32f10x::Pin('B', 0);
+    inline static sjsu::stm32f10x::Pin pb1 = sjsu::stm32f10x::Pin('B', 1);
+
+    // Timer 4
+    inline static sjsu::stm32f10x::Pin pb6 = sjsu::stm32f10x::Pin('B', 6);
+    inline static sjsu::stm32f10x::Pin pb7 = sjsu::stm32f10x::Pin('B', 7);
+    inline static sjsu::stm32f10x::Pin pb8 = sjsu::stm32f10x::Pin('B', 8);
+    inline static sjsu::stm32f10x::Pin pb9 = sjsu::stm32f10x::Pin('B', 9);
+
+   public:
+    inline static const PwmPin_t kPin_a_8 = {
```

Comment:
```suggestion
    inline static const PwmPin_t kA8 = {
```
This is a more concise name for the pin that still makes it obvious what you are doing.

Reply:
Although I think I need a new API for PWM in general to make it more explicit which pins go to which timers so that users don't have to go to the datasheet to know which pins go to which timers.